test/extended/networking/acl_audit_log.go

```diff
@@ -0,0 +1,213 @@
+	var err error
+
+	makeNamespaceScheduleToAllNodes(f)
+	nsNoACLLog, err := f.CreateNamespace("acl-logging-off", map[string]string{})
+	expectNoError(err)
+
+	ns[0] = f.Namespace.Name
+	ns[1] = nsNoACLLog.Name
+
+	nodes[0], nodes[1], err = findAppropriateNodes(f, DIFFERENT_NODE)
+	if err != nil {
+		return err
+	}
+
+	var pod, ip [3]string
+	var errs [3]error
+
+	// make the first two pods in ns acl-logging-on on node[0] and the other in acl-logging-off on node[1]
+	for i := range pod {
+		pod[i] = fmt.Sprintf("acl-logging-%d", i+1)
+		//pod := e2epod.NewAgnhostPod(f.Namespace.Name, podName, nil, nil, nil, "bash", "-c", "sleep 1000")
+		err := launchACLLoggingPod(f, ns[i/2], nodes[i/2].Name, pod[i])
+		expectNoError(err)
+		var zero int64
+		defer f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(context.Background(), pod[i], metav1.DeleteOptions{GracePeriodSeconds: &zero})
+	}
+	// make sure pods come up
+	for i := range pod {
+		ip[i], errs[i] = waitForACLLoggingPod(f, ns[i/2], pod[i])
+		expectNoError(errs[i])
+	}
+
+	// make two network policies, a default deny all and allow from same namespace in ns "acl-logging-on"
+	policy, err := makeDenyAllPolicy(f)
+	expectNoError(err)
+	defer cleanupNetworkPolicy(f, policy)
+
+	policy, err = makeFromSameNSPolicy(f)
+	expectNoError(err)
+	defer cleanupNetworkPolicy(f, policy)
+
+	// Ping pod[0] from pod[1] which should succeed since it's in the same namespace with out == ip[0]
+	// Should hit the `allow-same-namespace` networkpoliocy
+	out, _, err := oc.Run("exec").Args(pod[1], "--", "curl", "-m", "1", ip[0]+"/clientip").Outputs()
+	if !strings.Contains(out, ip[1]) || err != nil {
+		return err
+	}
+
+	// Ping pod[0] from pod[2] which will not succeed since it's in a different namespace
+	// Should hit the all-deny policy
+	out, _, err = oc.AsAdmin().Run("exec").Args(pod[2], "-n", ns[1], "--", "curl", "-m", "1", ip[0]+"/clientip").Outputs()
+	expectError(err)
+
+	// Ensure audit logs are there and that `oc adm node-logs` command adequately collects them
+	out, _, err = oc.AsAdmin().Run("adm").Args("node-logs", "--since=5s", nodes[0].Name, "--path=/ovn/acl-audit-log.log").Outputs()
+	expectNoError(err)
+
+	allowLogFound := false
+	denyLogFound := false
+
+	// Ensure the ACL audit logs are correct
+	for _, logLine := range strings.Split(out, "\n") {
+		if strings.Contains(logLine, "allow-from-same-ns\", verdict=allow") && strings.Contains(logLine, "nw_src="+ip[1]) &&
+			strings.Contains(logLine, "nw_dst="+ip[0]) {
+			allowLogFound = true
+		}
+
+		if strings.Contains(logLine, "default-deny-all\", verdict=drop") && strings.Contains(logLine, "nw_src="+ip[2]) &&
+			strings.Contains(logLine, "nw_dst="+ip[0]) {
+			denyLogFound = true
+		}
+
+		if allowLogFound && denyLogFound {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Acl logs exist but do not contain correct src dst ips and actions")
+}
+
+func launchACLLoggingPod(f *e2e.Framework, namespace string, nodeName string, podName string) error {
```

Comment:
We have access to the whole k8s library here; surely something must provide this function already....

Reply:
Unfortunately There isn't a default way to make a pod with the agnhost netexec image for linux, since `framework.pod.createpod` uses the busybox image by default instead.
 This is similar to what is done in the upstream network policy tests where they first 

create the spec manually 
https://github.com/kubernetes/kubernetes/blob/v1.21.1/test/e2e/network/netpol/model.go#L259

And then manually make the Pod 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/netpol/kubemanager.go#L179

Which is why I do the same thing here

Reply:
Arg I might have been wrong here, Let me try to use https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework/pod#NewAgnhostPod